src/frontend/state/GlobalState.tsx

```diff
@@ -395,11 +395,11 @@ class GlobalState extends PureComponent<Props> {
   }
 
   addGameToCustomCategory = (category: string, appName: string) => {
-    const newCustomCategories = this.state.customCategories
-
-    if (!newCustomCategories[category]) newCustomCategories[category] = []
+    const newCustomCategories = { ...this.state.customCategories }
```

Comment:
Wouldn't this line's change alone be enough to resolve the issue? I understand the issue is object equality related, so creating a new object to hold the categories should result in a re-render, no?

Reply:
I guess I tested it wrong haha, updated it to just change the new categories object and not the values references